services/server/server.go

```diff
@@ -255,11 +258,27 @@ func (s *Server) Version(ctx context.Context, req *serverpb.VersionRequest) (*se
 // Readiness returns an error when some PMM Server component is not ready yet or is being restarted.
 // It can be used as for Docker health check or Kubernetes readiness probe.
 func (s *Server) Readiness(ctx context.Context, req *serverpb.ReadinessRequest) (*serverpb.ReadinessResponse, error) {
-	// TODO https://jira.percona.com/browse/PMM-1962
 
-	if err := s.prometheus.Check(ctx); err != nil {
+	errs := make([]error, 0)
+	errs = append(errs, s.prometheus.Check(ctx))
+	errs = append(errs, s.grafanaClient.Check(ctx))
+
+	var err error
+	// concat the errors so the client will receive all the errors at once
+	for _, e := range errs {
+		if e != nil {
+			if err == nil {
+				err = e
+				continue
+			}
+			err = fmt.Errorf("%s, %w", err, e)
```

Comment:
We definitely don't want to start using `%w` – it is not compatible with pkg/errors, and migration is not trivial.